state/redis/redis.go

```diff
@@ -318,6 +340,19 @@ func (r *StateStore) parseETag(req *state.SetRequest) (int, error) {
 	return ver, nil
 }
 
+func (r *StateStore) parseTTL(req *state.SetRequest) int {
+	if val, ok := req.Metadata[ttlInSeconds]; ok && val != "" {
+		parsedVal, err := strconv.ParseInt(val, defaultBase, defaultBitSize)
+		if err != nil {
+			return 0
```

Comment:
Please, return error and fail the request. Swallowing an error and returning a valid number can hide a problem for in the application code.